kfs-ld/src/main/java/edu/arizona/kfs/module/ld/batch/service/impl/LaborEncumbranceAdjustmentServiceImpl.java

```diff
@@ -0,0 +1,626 @@
+                        outputStream.println( reversingOriginEntry.getLine() );
+                        jobData.outputCount++;
+                        jobData.removedBalanceCount++;
+                        // and advance the balance line iterator
+                        advanceBalanceFile = true;
+                    }
+                    
+                    
+                }
+                // depending on comparison, increment the appropriate counters
+                if (advanceEncumbranceFile) {
+                    jobData.inputLinesProcessed++;
+                    // move to the next line
+                    encumbranceLine = inputReader.readLine();
+                    lineNumber++;
+                }
+                if (advanceBalanceFile) {
+                    jobData.balanceLinesProcessed++;
+                    balanceLine = balanceReader.readLine();
+                }
+            }
+            outputStream.flush();
+            
+            try{
+            	//Create recon file.
+            	reconStream = new PrintStream(reconFile);
+            	reconStream.println("c ld_ldgr_entr_t " + jobData.outputCount +";");
+            	reconStream.println("s trn_ldgr_entr_amt " + reconAmount +";");
+            	reconStream.print("e 02;");            	
+            } catch ( IOException ex ) {
+                LOG.error( "Unable to create recon file for labor encumbrance adjustment output file.", ex );
+                throw new RuntimeException("Unable to create recon file for labor encumbrance adjustment output file.", ex);
+            }
+            
+            try {
+                String doneFileName = StringUtils.removeEnd(outputFile.getAbsolutePath(), ".data") + GeneralLedgerConstants.BatchFileSystem.DONE_FILE_EXTENSION;
+                File doneFile = new File( doneFileName );
+                doneFile.createNewFile();
+            } catch ( IOException ex ) {
+                LOG.error( "Unable to create done file for labor encumbrance adjustment output file.", ex );
+                throw new RuntimeException("Unable to create done file for labor encumbrance adjustment output file.", ex);
+            }
+        }
+        catch (FileNotFoundException ex) {
+            LOG.warn("Missing Input file:" + ex.getMessage(), ex);
+            return false;
+        }
+        catch (IOException ex) {
+            LOG.warn("Problem reading input files:" + ex.getMessage(), ex);
+            throw new RuntimeException("Error processing encumbrance balance file", ex);
+        }
+        finally {
+            IOUtils.closeQuietly(inputReader);
+            IOUtils.closeQuietly(balanceReader);
+            IOUtils.closeQuietly(outputStream);
+            IOUtils.closeQuietly(errorStream);
+            IOUtils.closeQuietly(reconStream);
+            if ( reportWriterService != null ) {
+                ((WrappingBatchService)reportWriterService).initialize();
```

Comment:
The casting is needed since multiple interfaces are implemented.  WrappingBatchService has initialize method.